src/Base/UnitsApi.cpp

```diff
@@ -139,23 +127,34 @@ void UnitsApi::setSchema(UnitSystem system)
                                        // Quantity (e.g. mi=1.8km rather then 1.6km).
 }
 
-QString UnitsApi::toString(const Base::Quantity& quantity, const QuantityFormat& format)
+std::string UnitsApi::toString(const Base::Quantity& quantity, const QuantityFormat& format)
 {
-    QString value = QString::fromLatin1("'%1 %2'")
-                        .arg(quantity.getValue(), 0, format.toFormat(), format.precision)
-                        .arg(quantity.getUnit().getString());
-    return value;
+    return '\'' + toNumber(quantity, format) + ' ' + quantity.getUnit().getString() + '\'';
 }
 
-QString UnitsApi::toNumber(const Base::Quantity& quantity, const QuantityFormat& format)
+std::string UnitsApi::toNumber(const Base::Quantity& quantity, const QuantityFormat& format)
 {
     return toNumber(quantity.getValue(), format);
 }
 
-QString UnitsApi::toNumber(double value, const QuantityFormat& format)
+std::string UnitsApi::toNumber(double value, const QuantityFormat& format)
 {
-    QString number = QString::fromLatin1("%1").arg(value, 0, format.toFormat(), format.precision);
-    return number;
+    std::stringstream ss;
+
+    switch (format.format) {
+        case QuantityFormat::Fixed:
+            ss << std::fixed;
+            break;
+        case QuantityFormat::Scientific:
+            ss << std::scientific;
+            break;
+        default:
+            break;
+    }
+    ss.imbue(std::locale());

```

Comment:
This changes the output of the string. Before it was independent of the locale and now it will depend on it.

Reply:
Fixed (now I'm not sure why I added locale awareness, I remember I was searching in documentation how to do it for `stringstream`).